contents/javascript/Javascript-framework-Angular.md

```diff
@@ -0,0 +1,226 @@
+
+This [article](https://medium.com/@preethi.s/angular-custom-two-way-data-binding-3e618309d6c7) on Medium provides a good introduction to two-way data binding in Angular.
+
+### Disadvantages of Angular.
+
+Like any other framework/library, Angular has its share of disadvantages.
+
+1. **Steep Learning Curve** - Angular requires you to learn many concepts, such as directives, modules, components, services and many more. 
+The large number of new concepts can be confusing to newcomers. Additionally, Angular requires you to learn a new language, TypeScript.
+
+2. **Opinionated Framework** - Angular is opinionated about how you structure your code. This means that Angular provides you defaults for building applications. For instance, there are in-built services to perform data fetching, state management and much more. It is also opinionated about the development language as it requires you to use TypeScript.<br/>
+In contrast, React allows you to integrate any third party services in your application.
+    
+### Angular when compared with other popular frontend frameworks
+
+[React]("https://reactjs.org/") and [Vue]("https://vuejs.org/") are two popular alternatives to Angular.
+
+React is a popular Javascript library, open sourced by Facebook. The ease of learning React is a key advantage.
+It also provides more flexibility as it allows integration of third party libraries.
+Similarly, Vue offers some advantages, like flexibility, simple structure and ease of integration.
+
+Every framework has its own pros and cons. The framework you choose depends on the requirements of the application. Below are a few resources that give a brief comparison between the popular frontend frameworks.
+
+1. [React vs Angular vs Vue](https://medium.com/@TechMagic/reactjs-vs-angular5-vs-vue-js-what-to-choose-in-2018-b91e028fa91d) - A brief comparison between the three most popular frontend tools.
+2. [Why use Vue](https://medium.com/@brainmobi/advantages-of-using-vue-js-for-your-web-applications-7e460cadfffc) - Advantages of using Vue.
+3. [Angular vs React](https://programmingwithmosh.com/react/react-vs-angular/) - A comprehensive comparison between React and Angular.
+
+## How to get started with Angular?
+
+Given below are a few steps that will help you ease into the Angular environment.
+
+1. [Different versions of Angular](https://www.simplilearn.com/angularjs-vs-angular-2-vs-angular-4-differences-article) - Before you start learning Angular, you must have a clear idea of its different versions.
+
+2. [Start learning TypeScript](https://www.typescriptlang.org/docs/home.html) - Development in Angular relies heavily on TypeScript. You can start learning the basics of the new language from its official website.
+
+3. [The official Angular website](https://angular.io/guide/quickstart) - offers a good tutorial to get started. It shows you how to build and run a simple Hello World application in Angular.
```

Comment:
you wan to add double quotation to Hello World?

Reply:
Yeah makes sense, thanks for pointing it out!